File: Serene1/Serene1/Serene1.Script/Imports/ServerImports/Northwind.CustomerDemographicForm.cs
﻿using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene1.Northwind
{
    public partial class CustomerDemographicForm : PrefixedContext
    {
        [InlineConstant]
        public const string FormKey = "Northwind.CustomerDemographic";

        public CustomerDemographicForm(string idPrefix) : base(idPrefix) { }

        public StringEditor CustomerTypeID { [InlineCode("{this}.w('CustomerTypeID', Serenity.StringEditor)")] get; private set; }
        public StringEditor CustomerDesc { [InlineCode("{this}.w('CustomerDesc', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene1/Serene1/Serene1.Web/Modules/Common/Imports/ServerTypings/Northwind.CustomerDemographicForm.ts
﻿namespace Serene1.Northwind {
    export class CustomerDemographicForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.CustomerDemographic';

    }

    export interface CustomerDemographicForm {
        CustomerTypeID(): Serenity.StringEditor;
        CustomerDesc(): Serenity.StringEditor;
    }

    [['CustomerTypeID', Serenity.StringEditor], ['CustomerDesc', Serenity.StringEditor]].forEach(x => CustomerDemographicForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

